### PR TITLE
phone-number: add missing edge cases

### PR DESCRIPTION
**Pull request by cmccandless**
The existing descriptions for invalid area codes and exchange codes both cite a range, but only test a single value. This PR tests for all values outside the valid range.

### CODE REVIEW

**Comment by rpottsoh** [comment]
Thanks for bumping that.  It is so easy to miss.

File: exercises/phone-number/canonical-data.json [hunk 1]
```diff
@@ -72,15 +72,37 @@
...
+              "expected": null
+            },
+            {
+              "description": "invalid if area code starts with 1",
+              "property": "clean",
+              "phrase": "(123) 456-7890",
+              "expected": null
+            }
+          ]
         },
```
Comment (leenipper) [inline exercises/phone-number/canonical-data.json]: Agree with the new edge test cases.  But, would prefer it without another level of "cases": [ ].  I suggest the two invalid cases be at same level as the other clean test cases.  Otherwise, the change will break some generators unnecessarily.

File: exercises/phone-number/canonical-data.json [hunk 2]
```diff
@@ -72,15 +72,37 @@
...
+              "phrase": "(223) 056-7890",
+              "expected": null
+            },
+            {
+              "description": "invalid if exchange code starts with 1",
+              "property": "clean",
+              "phrase": "(223) 156-7890",
+              "expected": null
+            }
+          ]
```
Comment (leenipper) [inline exercises/phone-number/canonical-data.json]: Ditto here. Prefer two new invalid cases, but without another nested level of cases.

File: exercises/phone-number/canonical-data.json (hunk 1 above)
Comment (cmccandless) [reply exercises/phone-number/canonical-data.json]: Fair enough; done.

Comment (leenipper) [reply exercises/phone-number/canonical-data.json]: Thanks!

**Comment by ErikSchierboom** [comment]
Merged. Thanks!
